File: src/location/finders/LocationFinder.java
package location.finders;

import location.RecordManager;

/**
 * Created by zsmb on 2016-07-23.
 */
public abstract class LocationFinder {

    final RecordManager rm;

    LocationFinder(RecordManager rm) {
        this.rm = rm;
    }

    /**
     * Returns the location for the specified time
     *
     * @param timeMS the time to check
     * @return the location in a two-length array, latitude(0) and longitude(1)
     */
    public abstract double[] getLocation(long timeMS);

    /**
     * Prints statistics about the matches found on the standard output
     */
    public abstract void printStats();
}

File: src/main/kotlin/location/finders/LocationFinder.kt
package location.finders


import location.RecordStore

/**
 * Abstract base class of location finders
 * These classes find you a GPS location based on a given timestamp,
 * using a RecordStore to get access to location records
 */
abstract class LocationFinder internal constructor(protected val recordStore: RecordStore) {

    /**
     * Returns the location (lat, long) for the specified time
     */
    abstract fun getLocation(timeMS: Long): Pair<Double, Double>?

    /**
     * Prints statistics about the matches found on the standard output
     */
    abstract fun printStats()

}
